localstack/services/kinesis/kinesis_starter.py

```diff
@@ -21,9 +31,49 @@ def appy_patches():
 
 
 def start_kinesis(port=None, asynchronous=False, update_listener=None):
+    if KINESIS_PROVIDER == 'kinesis-mock':
+        return start_kinesis_mock(port=port, asynchronous=asynchronous, update_listener=update_listener)
+    if KINESIS_PROVIDER == 'kinesalite':
+        return start_kinesalite(port=port, asynchronous=asynchronous, update_listener=update_listener)
+    raise Exception('Unsupported Kinesis provider "%s"' % KINESIS_PROVIDER)
+
+
+def start_kinesis_mock(port=None, asynchronous=False, update_listener=None):
+    target_dir = os.path.join(INSTALL_DIR_INFRA, 'kinesis-mock')
+
+    if config.is_in_docker:
+        target_file_name = 'kinesis-mock-linux-amd64-static'
```

Comment:
This will use the static distribution for the linux native-image executable if we are in the localstack docker image (tested in localstack/localstack). Native Image executables are unfortunately too specific to the architectures to be trusted properly in a non-docker distribution, so we'll use the jar in that scenario.

Reply:
It is still possible to try to use the native-image executables outside of docker _if_ we can identify that the environment is amd64. I'm not terribly familiar with how to do this in Python. I know of ways to check if its 64-bit and which OS is running but not the arch itself. Are you aware of a way to do this @whummer ?

Reply:
I think I figured this out. resolving.